m4opt/synphot/background/_cerenkov/__init__.py

```diff
@@ -0,0 +1,339 @@
+from dataclasses import dataclass
+from typing import Literal
+
+import numpy as np
+from aep8 import flux
+from astropy import units as u
+from astropy.constants import c, h, m_e, m_p
+from astropy.table import Table
+from scipy.interpolate import CubicSpline
+from synphot import Empirical1D, SourceSpectrum
+
+from .._core import BACKGROUND_SOLID_ANGLE
+from ._electron_loss import get_electron_energy_loss
+from ._refraction_index import get_refraction_index
+
+
+def radiation_belt_flux_table(
+    observer_location,
+    obstime,
+    energy: tuple[u.Quantity, u.Quantity] = (0.05 * u.MeV, 8.5 * u.MeV),
+    nbins: int = 20,
+    particle: Literal["e", "p"] = "e",
+    solar: Literal["max", "min"] = "max",
+) -> Table:
+    """
+    Returns a table of integral flux values from Earth's radiation belts.
+    """
+    emin, emax = energy
+    energy_bins = np.geomspace(emin, emax, num=nbins)
+    flux_integral = [
+        flux(
+            observer_location,
+            obstime,
+            e,
+            kind="integral",
+            solar=solar,
+            particle=particle,
+        )
+        for e in energy_bins
+    ]
+    return Table([energy_bins, u.Quantity(flux_integral)], names=["energy", "flux"])
+
+
+@dataclass
```

Comment:
Why is this a class?

Reply:
@lpsinger  The way I am calling `CerenkovEmission` and `CerenkovBackground` requires them to use the @dataclass decorator. Without @dataclass, those keyword argument initializations wouldn’t work.